assets/javascripts/discourse/components/chat-composer.js

```diff
@@ -341,7 +341,23 @@ export default Component.extend(TextareaTextManipulation, ComposerUploadUppy, {
           const full = `:${term}`;
           term = term.toLowerCase();
 
-          if (term.length < this.siteSettings.emoji_autocomplete_min_chars) {
+          // We need to avoid quick emoji autocomplete cause it can interfere with quick
+          // typing, set minimal length to 2
+          let minLength = Math.max(
+            this.siteSettings.emoji_autocomplete_min_chars,
+            2
+          );
+
+          if (term.length < minLength) {
+            return resolve(SKIP);
+          }
+
+          // bypass :-p and other common typed smileys
+          if (
+            !term.match(
+              /[^-\{\}\[\]\(\)\*_\<\>\\\/].*[^-\{\}\[\]\(\)\*_\<\>\\\/]/
```

Comment:
Sorry just a follow up comment but I think having test cases for this regexp is important. There is a comment left but common typed smileys is loosely defined and figuring out what the regexp is trying to do is not straight forward.

Reply:
testing autocomplete is very difficult at the moment, very flaky, we need to fix that.

Reply:
I don’t mean we should test autocomplete but the matching of the regexp.